appcues/src/main/java/com/appcues/analytics/ExperienceLifecycleEvent.kt

```diff
@@ -120,7 +120,7 @@ internal sealed class ExperienceLifecycleEvent(
                     hashMapOf(
                         INTERACTION_TYPE_KEY to interactionType.analyticsName(),
                         INTERACTION_DATA_KEY to when (interactionType) {
-                            FORM_SUBMITTED -> step.formState
+                            FORM_SUBMITTED -> step.formState.toAnalyticsMap()
```

Comment:
so, doing it this way will break the debugger.  See https://github.com/appcues/appcues-android-sdk/blob/main/appcues/src/main/java/com/appcues/debugger/DebuggerRecentEventsManager.kt#L208-L214 where it tries to pull out the `ExperienceStepFormState` from the property map.

That is why it was done that way originally at least. There are likely multiple ways we could handle that.  But this is why on iOS it was done _only_ at the listener level, and not further upstream like this. https://github.com/appcues/appcues-ios-sdk/pull/253/files.  We may also need to handle the Date mapping similarly, as the goal was to move all that logic into the core SDK rather than x-plat wrappers.

side track....
@mmaatttt  I notice now the iOS update maps Date to Int64, but in the wrappers we use Double types
[capacitor android](https://github.com/appcues/appcues-capacitor-plugin/blob/main/android/src/main/java/com/appcues/sdk/capacitor/AppcuesPlugin.kt#L60)
[capacitor ios](https://github.com/appcues/appcues-capacitor-plugin/blob/main/ios/Plugin/AppcuesPlugin.swift#L174)
[react native ios](https://github.com/appcues/appcues-react-native-module/blob/main/ios/AppcuesReactNative.swift)

I don't fully recall - but do those not handle Int64/long and we should use double there as well?

Reply:
this was the PR where the form state handling was originally added to simplify that debugger usage, for reference https://github.com/appcues/appcues-android-sdk/pull/234

Reply:
I did not connect the dots before, but this and the other open PR are related in regards to the debugger. what I propose is to map here and fix the debugger to support more generally hashmap enteries for this Interaction Data instead of casting it to ExperienceStepFormState. this way we can make it work for all types of interactions

Reply:
I think it could be done, but it will result in having some messier code dealing with nested map indexing in the debugger.  That was the motivation for handling with typed property objects previously.  This was an intermediate commit where I was doing that https://github.com/appcues/appcues-android-sdk/blob/e6988f14241f4f74f1d6048ec4b9eb4ebe73e9bc/appcues/src/main/java/com/appcues/debugger/DebuggerRecentEventsManager.kt#L169-L182 -- it can work, just a big messier with the map indexing, and I was hoping to avoid that.

Reply:
Fixing it to be `Double` in iOS: https://github.com/appcues/appcues-ios-sdk/pull/257